src/local_scheduler/local_scheduler.cc

```diff
@@ -828,7 +828,9 @@ void process_message(event_loop *loop,
     if (ActorID_equal(TaskSpec_actor_id(spec), NIL_ACTOR_ID)) {
       handle_task_submitted(state, state->algorithm_state, spec, length);
     } else {
-      handle_actor_task_submitted(state, state->algorithm_state, spec, length);
+      TaskSpec *spec_copy = (TaskSpec*) malloc(utarray_len(state->input_buffer));
+      memcpy(spec_copy, spec, utarray_len(state->input_buffer));
+      handle_actor_task_submitted(state, state->algorithm_state, spec_copy, length);
```

Comment:
instead of having `handle_actor_task_submitted` take ownership of the task spec, let's have it copy the task spec inside.